### DISCUSSION TITLE
Implement (with a warning) sensitivity methods using finite differences

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1575] [Ideas/Science]
For sensitivities, this requires 2x number of parameters per iteration if we use central differences; 1x the number of parameters if we use forward/backward differences.

For 2nd order sensitivities, more function evaluations are required.

**Comment by ben18785** [comment]
[This](http://www.mathematik.uni-dortmund.de/~kuzmin/cfdintro/lecture4.pdf) is a good reference on finite differences.

Note that I don't expect this approach to be as computationally efficient as using automatic differentiation or calculating the sensitivities outside of Pints. It will, however, allow users to quantify, in terms of the number of effective samples per _iteration_ not _computational time_ whether it is worth solving for sensitivities.

**Comment by MichaelClerx** [comment]
I have some serious doubts about this! 

Apart from my gut feeling (backed up by some bad experiences) that they're a terrible idea for serious problems, there's the question of where do we stop with Pints.
Yes it's something we _could_ add, but I'm much happier saying "pints is for optimising / sampling" and leaving the rest for the users and other specialist packages. (See #501 )

**Comment by MichaelClerx** [comment]
This is related to #101

**Comment by MichaelClerx** [comment]
Example of how this could be implemented in a wrapper: https://github.com/pints-team/pints/pull/892
